File: app/src/main/java/com/dakota/gallimore/portfolio/ChildEducation.java
package com.dakota.gallimore.portfolio;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by Dakota on 10/26/2017.
 */

public class ChildEducation implements Parcelable {

    public static final Creator<ChildEducation> CREATOR = new Creator<ChildEducation>() {
        @Override
        public ChildEducation createFromParcel(Parcel in) {
            return new ChildEducation(in);
        }

        @Override
        public ChildEducation[] newArray(int size) {
            return new ChildEducation[size];
        }
    };
    private String header;
    private String date;
    private String details;

    public ChildEducation(String headerIn, String dateIn, String detailsIn) {
        header = headerIn;
        date = dateIn;
        details = detailsIn;
    }

    protected ChildEducation(Parcel in) {
        header = in.readString();
        date = in.readString();
        details = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(this.header);
        parcel.writeString(this.date);
        parcel.writeString(this.details);
    }

    public String getHeader() {
        return header;
    }

    public void setHeader(String header) {
        this.header = header;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }

}

File: app/src/main/java/com/dakota/gallimore/portfolio/ChildEducation.kt
package com.dakota.gallimore.portfolio

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by Dakota on 10/26/2017.
 */

class ChildEducation : Parcelable {
    var header: String? = null
    var date: String? = null
    var details: String? = null

    constructor(headerIn: String, dateIn: String, detailsIn: String) {
        header = headerIn
        date = dateIn
        details = detailsIn
    }

    protected constructor(`in`: Parcel) {
        header = `in`.readString()
        date = `in`.readString()
        details = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(this.header)
        parcel.writeString(this.date)
        parcel.writeString(this.details)
    }

    companion object {

        val CREATOR: Parcelable.Creator<ChildEducation> = object : Parcelable.Creator<ChildEducation> {
            override fun createFromParcel(`in`: Parcel): ChildEducation {
                return ChildEducation(`in`)
            }

            override fun newArray(size: Int): Array<ChildEducation?> {
                return arrayOfNulls(size)
            }
        }
    }

}
